### ISSUE TITLE
dev.py does not propagate error codes, thus hides errors on CI

### ISSUE THREAD

**Issue by ev-br** [scipy/scipy#16476]
CI providers rely on exit codes to report whether tests fail or pass (e.g. https://docs.github.com/en/actions/creating-actions/setting-exit-codes-for-actions)

However the click-based dev.py in some cases reports the exit code 0 (meaning success) even if the subtask exit code is 1 (meaning a failure).

To reproduce, add an error to a docstring example:

```diff
$ git diff
diff --git a/scipy/io/_harwell_boeing/hb.py b/scipy/io/_harwell_boeing/hb.py
index bab0f242fa..15a0896270 100644
--- a/scipy/io/_harwell_boeing/hb.py
+++ b/scipy/io/_harwell_boeing/hb.py
@@ -553,6 +553,9 @@ def hb_write(path_or_open_file, m, hb_info=None):
       (1, 1)   1.0
       (2, 2)   1.0
 
+    >>> 2/3
+    42
+
     """
     m = m.tocsc(copy=False)
 
```

then, run

```
(scipy-dev) br@gonzales:~/repos/scipy/scipy$ python dev.py refguide-check -s io
💻  ninja -C /home/br/repos/scipy/scipy/build
ninja: Entering directory `/home/br/repos/scipy/scipy/build'
[2/2] Generating scipy/generate-config with a custom command
Build OK
💻  meson install -C build
Installing, see meson-install.log...
Installation OK
Running checks for 1 modules:
scipy.io ....................F.........

========
scipy.io
========

scipy.io.hb_write
-----------------

<doctest hb_write[0]>:1: DeprecationWarning: Please use `hb_read` from the `scipy.io` namespace, the `scipy.io.harwell_boeing` namespace is deprecated.
  from scipy.io.harwell_boeing import hb_read, hb_write
<doctest hb_write[0]>:1: DeprecationWarning: Please use `hb_write` from the `scipy.io` namespace, the `scipy.io.harwell_boeing` namespace is deprecated.
  from scipy.io.harwell_boeing import hb_read, hb_write


File "build-install/lib/python3.8/site-packages/scipy/io/_harwell_boeing/hb.py", line 556, in hb_write
Failed example:
    2/3
Expected:
    42
Got:
    0.6666666666666666


ERROR: refguide or doctests have errors
(scipy-dev) br@gonzales:~/repos/scipy/scipy$ echo $?
0
```

Note that the exit code is zero, meaning success.

However, running refguide-check directly does report an error exit code:

```
(scipy-dev) br@gonzales:~/repos/scipy/scipy$ python dev.py shell

<cd to the repo root>

(scipy-dev) br@gonzales:~/repos/scipy/scipy$ 
br@gonzales:~/repos/scipy/scipy$ python tools/refguide_check.py --doctests io
Running checks for 1 modules:
scipy.io ....................F.........

========
scipy.io
========

scipy.io.hb_write
-----------------

<doctest hb_write[0]>:1: DeprecationWarning: Please use `hb_read` from the `scipy.io` namespace, the `scipy.io.harwell_boeing` namespace is deprecated.
  from scipy.io.harwell_boeing import hb_read, hb_write
<doctest hb_write[0]>:1: DeprecationWarning: Please use `hb_write` from the `scipy.io` namespace, the `scipy.io.harwell_boeing` namespace is deprecated.
  from scipy.io.harwell_boeing import hb_read, hb_write


File "build-install/lib/python3.8/site-packages/scipy/io/_harwell_boeing/hb.py", line 556, in hb_write
Failed example:
    2/3
Expected:
    42
Got:
    0.6666666666666666


ERROR: refguide or doctests have errors
br@gonzales:~/repos/scipy/scipy$ echo $?
1
```

**Comment by schettino72** [comment]
I released pydevtool 0.3 fixing this issue.
conda-forge updates automatically? after that just needs to bump the version requirement on `environment.yml`.

**Comment by rgommers** [comment]
Awesome, thanks @schettino72. Yes, conda-forge updates automatically, the bot will open a PR within max 6 hours I believe.
